### ISSUE TITLE
(3.0.89) `Jadeite>>Auto Commit enabled` menu out of sync with auto commit state

### ISSUE THREAD

**Issue by dalehenrich** [ericwinger/Jade#650]
Note that the `Jadeite` menu does not have the `Auto Commit enabled` menu item marked, while the icon at the bottom of the project browser is green:
![image](https://user-images.githubusercontent.com/623951/69768317-f5523b00-1134-11ea-8c06-919ec512bf76.png)
I've been in habit of using the menu item to "tell me if autocommit is enabled" because the autocommit icon is not present in all windows and I don't always have a project browser open ... so the menu probably should be accurate

**Comment by LisaAlmarode** [comment]
I saw something like this - #620 - but it's all working correctly for me in 3.0.89.  Did you set this via preferences?

**Comment by dalehenrich** [comment]
I have autocommit enabled in my prefs file ... and I tried to use the menu to turn it on/off(???), but the menu just stayed with it unselected ... I've seen this ever since I started using prefs file(?) and it wasn't until today that I remembered to look at the icon ... I was attempting to toggle the menu item but it stayed stuck off --- I did notice that the icon was toggling correctly ... but the menu was just stuff `off` ...

Here's my prefs file:
```
#Remove # to uncomment option. 
#legal values are either a boolean or a name

autocommitEnabled: true
#autocompleteEnabled: false
#breakpointsEnabled: true
#loggingEnabled: false
#openBrowserOnClass: <class name>
openBrowserOnPackage: Rowan-TestsV2
#openBrowserOnProject: <project name> 
openSUnitBrowserOnPackage: Rowan-TestsV2
#openWorkspace: <file path without quotes or nil> 
transcriptEnabled: false
```

**Comment by LisaAlmarode** [comment]
This doesn't happen to me under normal circumstances.  
However, I played around a bit with manual transaction mode, and after I confirmed you'd get bigger breakage than you report :-) , and restored normal mode... I'm now seeing the autocommit can be toggled from the green camera icon or the menu, but the menu bar refused to add a black dot. 

This seems to reproduce the problems in 3.0.90: 
- login, using Dale's prefs file.
- toggle autocommit so it's off
- in console workspace, execute ```System transactionMode: #manualBegin```
- look at jadeite menu to check icon status.
- in console workspace, execute ```System transactionMode: #autoBegin```

Now, the autocommit black dot is no longer shown, although it's otherwise functional.

**Comment by ericwinger** [comment]
@LisaAlmarode AutoCommit is a Jadeite-specific feature. Is the request that we synchronize this with the transaction mode on the server?  

If we want that, I'd recommend we open a new issue to provide Jadeite gui tools for manipulating the transaction mode and expect the user to use them. (Ad-hoc executions through the workspace are harder to keep in sync with the gui). Once we have those tools in place, we can manipulate auto commit as necessary but the rules should be spelled out in an issue. 

I can't reproduce the reported issue in 3.0.92pre-release1.

**Comment by LisaAlmarode** [comment]
It's not the transaction mode; any kind of execute in the workspace breaks the icon in the list.
- turn off autocommit
- do an execute
- turn on autocommit
- experience no icon in menu, although camera is green

**Comment by ericwinger** [comment]
Turns out the execution had nothing to do with the problem. Simply selecting the Workspace tab in the console did it. 

When menu items are displayed, each item is passed to every presenter in the window. If the presenter implements the menu command, the presenter is asked to check the menu item. In the case of the autocommit menu item command, the presenter for the workspace, `CodeSourcePresenter` also implemented the same command - `flipAutoCommit`. Therefore, menu item checking was done in the wrong presenter. 

This is fixed in 3.0.92pre-release2.

**Comment by ericwinger** [comment]
fyi - This would have been a problem in some of the other console tabs as well.
